internal/utils/utils_test.go

```diff
@@ -0,0 +1,192 @@
+package utils_test
+
+import (
+	"testing"
+
+	promoterv1alpha1 "github.com/argoproj-labs/gitops-promoter/api/v1alpha1"
+	"github.com/argoproj-labs/gitops-promoter/internal/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+const (
+	namespace           = "test-namespace"
+	controllerNamespace = "controller-namespace"
+)
+
+func TestGetScmProviderFromGitRepository(t *testing.T) {
```

Comment:
Can we turn these tests into ginkgo tests see: template_test.go, that way we get [coverage stil](https://github.com/argoproj-labs/gitops-promoter/pull/161) and we standardize testing.

Reply:
Done!